editor/components/block-drop-zone/index.js

```diff
@@ -53,10 +53,32 @@ function BlockDropZone( { index, isLocked, ...props } ) {
 		}
 	};
 
+	const onDrop = ( event, position ) => {
```

Comment:
For a component which is rendered once-per-block, I wish we were more conscious about the performance implications of creating new function references on every render. (Not entirely related to this pull request, since the pattern has already been established here).